chromium_src/net/socket/socks5_client_socket_auth.cc

```diff
@@ -0,0 +1,149 @@
+// Copyright 2017 The Brave Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "net/base/io_buffer.h"
+#include "net/socket/socks5_client_socket_auth.h"
+
+namespace net {
+
+SOCKS5ClientSocketAuth::SOCKS5ClientSocketAuth(
+    std::unique_ptr<ClientSocketHandle> transport_socket,
+    const HostResolver::RequestInfo& req_info,
+    const NetworkTrafficAnnotationTag& traffic_annotation,
+    const HostPortPair& proxy_host_port)
```

Comment:
I'm sure we talked about doing a subclass/friend for HostPortPair, but I can't remember why we didn't. The only place we create the auth version is in our own code so it seems like it should work?

Reply:
Would presumably require making several of HostPortPair's member functions virtual.  Not sure what the tradeoff is of the extra members vs making those functions virtual is.  Originally I didn't touch HostPortPair at all, but rather put it into ProxyServer.